mmv1/third_party/terraform/resources/resource_storage_bucket_object.go

```diff
@@ -288,6 +311,10 @@ func resourceStorageBucketObjectCreate(d *schema.ResourceData, meta interface{})
 	insertCall.Name(name)
 	insertCall.Media(media)
 
+	// This is done late as we need to add headers to enable customer encryption
+	if v, ok := d.GetOk("customer_encryption"); ok {
+		setEncryptionHeaders(v.(string), insertCall.Header())
```

Comment:
instead of setting headers lets set the `CustomerEncryption` value on `object` to an instance of `*ObjectCustomerEncryption`.

Reply:
I've tried that and it doesn't work, this object does not 'make' data encrypted, note that it doesn't even have 'key' attribute, only it's SHA256 sum.

I am not 100% sure what does it actually do.
I found that setting headers is the only way (that I could make it work). 
Also, @rileykarson here: https://github.com/hashicorp/terraform-provider-google/issues/8133 also mentioned it needs headers configured.

Reply:
weird ! no problem !